### PR TITLE
Feat: Add Blocks view on Milestone payload

### PR DESCRIPTION
**Pull request by msarcev**
# Description of change

- [x] Add backend support to fetch from chronicle API for milestone data
- [x] Test that the backend chronicle fetch works (needs updated Chronicle instance)
- [x] Tabify the milestone page (check/reuse TabbedSection that introduced for Address page)
- [x] Add one more tab with the list of Blocks (check figma below)
- [x] Adapt the designs of Block pages (all versions) to figma designs

Resolves https://github.com/iotaledger/explorer/issues/495

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [] My code follows the contribution guidelines for this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by msarcev** [comment]
Removed the advanced button toggle on Block pages (there is only rarely rendered ReceiptPayload left that uses it, which was hardcoded to "true" from Block pages side).
For the nonce, i'm not sure about moving it to metadata as technically is not part of the blockMetadata but it's on the Block itself, so i left it where it is.

File: api/src/routes/stardust/milestone/blocks/get.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * @returns The response.
+ */
+export async function get(
+    _: IConfiguration,
+    request: IMilestoneBlocksRequest
+): Promise<IMilestoneBlocksResponse> {
+    const networkService = ServiceFactory.get<NetworkService>("network");
+    const networks = networkService.networkNames();
+    ValidationHelper.oneOf(request.network, networks, "network");
+    ValidationHelper.number(Number(request.milestoneId), "milestoneId");
```
[the remark this answers is not in the document]
Comment (msarcev) [reply api/src/routes/stardust/milestone/blocks/get.ts:23]: Yes, cause it was shamelessly copied from a different route 😅

File: client/src/app/components/stardust/section/referenced-blocks/ReferencedBlocksSection.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+const PAGE_SIZE: number = 10;
+
+const ReferenceBlocksSection: React.FC<ReferenceBlocksSectionProps> = ({ blockIds }) => {
+    const isMounted = useRef(false);
+    const [currentPage, setCurrentPage] = useState<string[]>([]);
+    const [pageNumber, setPageNumber] = useState(1);
+
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (msarcev) [reply client/src/app/components/stardust/section/referenced-blocks/ReferencedBlocksSection.tsx:18]: Remove isMounted for this component. Added a task to add a custom hook for the functionality.

File: client/src/app/routes/stardust/Block.tsx [hunk 3]
```diff
@@ -165,14 +180,51 @@ const Block: React.FC<RouteComponentProps<BlockProps>> = (
...
+    const isMilestoneBlock = block?.payload?.type === MILESTONE_PAYLOAD_TYPE;
+    const isTransactionBlock = block?.payload?.type === TRANSACTION_PAYLOAD_TYPE;
     const isLinksDisabled = metadata?.ledgerInclusionState === "conflicting";
     const isLoading = Array.from(jobToStatus.values()).some(status => status !== PromiseStatus.DONE);
-    const milestoneId = block?.payload?.type === MILESTONE_PAYLOAD_TYPE ?
-        milestoneIdFromMilestonePayload(block.payload) : undefined;
+    const milestoneId = isMilestoneBlock ?
+        milestoneIdFromMilestonePayload(block.payload as IMilestonePayload) : undefined;
+    let pageTitle = "Block";
+    switch (block?.payload?.type) {
```
[the remark this answers is not in the document]
Comment (msarcev) [reply client/src/app/routes/stardust/Block.tsx:204]: Fixed on the one below (349) as it was exactly displaying the payload type.
As for the title i'm leaving it as we render different strings an it's only used in this component atm.

File: client/src/app/components/ModalProps.tsx [hunk 4]
```diff
@@ -1,3 +1,13 @@
+export interface ModalData {
+    title?: string;
+    description: string;
+    links?: {
+        label?: string;
+        href?: string;
+        isExternal?: boolean;
+    }[];
```
[the remark this answers is not in the document]
Comment (msarcev) [reply client/src/app/components/ModalProps.tsx:8]: The array is for `ModalData.links?` here, not `ModalData` itself which is still a single object.

File: client/src/helpers/stardust/transactionsHelper.ts [hunk 5]
```diff
@@ -171,10 +171,10 @@ export class TransactionsHelper {
      */
     public static buildIdHashForAliasOrNft(aliasOrNftId: string, outputId: string): string {
         return !HexHelper.toBigInt256(aliasOrNftId).eq(bigInt.zero) ?
-        aliasOrNftId :
-        HexHelper.addPrefix(
-            Converter.bytesToHex(Blake2b.sum256(Converter.hexToBytes(HexHelper.stripPrefix(outputId))))
-        );
+            aliasOrNftId :
+            HexHelper.addPrefix(
+                Converter.bytesToHex(Blake2b.sum256(Converter.hexToBytes(HexHelper.stripPrefix(outputId))))
```
[the remark this answers is not in the document]
Comment (msarcev) [reply client/src/helpers/stardust/transactionsHelper.ts]: Here i didn't make any actual changes (just used formatter :)), but sure, nice catch.

**Review by brancoder — APPROVED** [review]
LGTM

**Pull request merged**